henryPuma/assignments/slidingwindow/lc209/lc209.js

```diff
@@ -1,5 +1,34 @@
+
+// S: int > 0
+// Arr: arr.length >= 0
 function smallest_subarray_with_given_sum(s, arr) {
+  // Arr.length == 0
+  if (!arr.length) return 0;
+  // let startWindow: the start of a subarray
+  // let i: index of the current value in our Array
+  
+  // let arraySum : Store the overall sum of our sub array
+  // let minLength : the minimum length of our subarray where arraySum >= s
+  // let subLength: the length of the subarray
+  // [1 ,3 ,4, 4, 7]     s=5
   // TODO: Write code here
+  let startWindow=0, i=0, arraySum=0, subLength=0;
+
+  let minLength =  Number.MAX_SAFE_INTEGER;
+  const arrLength = arr.length;
+ 
+  while( i < arrLength){
+    arraySum += arr[i];
+    subLength +=1; 
+    if (arraySum >= s){
+      minLength = subLength < minLength ? subLength : minLength;
+      arraySum -= arr[startWindow];
+      startWindow += 1;
+      subLength -= 1;
+    }
+    i+=1;
+  }
+  return subLength;
 }
```

Comment:
Good work, love that you are initiating variables on same line. Be a tad bit careful that your interviewer unstands that you are being clever here.

Good use of ternary, again be mindful of the interviewer (say aloud "I am going to use a ternary here as opposed to a traditional if/else is that cool??"

Make sure to type it twice right away! Get those reps in! #Grind #GreatWork #POG